src/types/utils/EmptyObject.ts

```diff
@@ -7,7 +7,7 @@ function isNotEmptyObject<T extends Record<string, unknown> | Falsy>(arg: T | Em
     return Object.keys(arg ?? {}).length > 0;
 }
 
-function isEmptyObject<T>(obj: T): boolean {
+function isEmptyObject<T>(obj: T | EmptyObject | null | undefined): obj is EmptyObject {
```

Comment:
This is not entirely true, when you pass null/undefined this function returns true but object isn't an EmptyObject

Reply:
@blazejkustra Good point! Updated, let me know what you think